src/OpenTelemetry/Batch.cs

```diff
@@ -31,12 +32,14 @@ namespace OpenTelemetry
     {
         private readonly T item;
         private readonly CircularBuffer<T> circularBuffer;
+        private readonly T[] metrics;
         private readonly long targetCount;
 
         internal Batch(T item)
         {
             this.item = item ?? throw new ArgumentNullException(nameof(item));
             this.circularBuffer = null;
+            this.metrics = null;
```

Comment:
Just to call out loud here, do we need to initialize these to null, or they are null by default?

Reply:
Since `Batch` is a struct, all fields must be explicitly initialized.

Reply:
I see, thanks!

Reply:
@CodeBlanch I guess a private Batch `ctor` which sets everything to null plus aggressive inline, then reuse it across all other `internal` ctors?

Reply:
we cannot define a default ctor like `Batch()`.

Reply:
Maybe something like a default copy constructor?

Reply:
#3484